syntax = "proto3";

package ru.yandex;

message EchoRequest {
    string message = 1;
}

message EchoResponse {
    string message = 1;
}

service EchoService {
    /*
     * Returns echoed request message adding string "Echo: " in front of it.
     */
    rpc getEcho(EchoRequest) returns (EchoResponse) {}

    /*
     * Starts bidirectional echo stream.
     */
    rpc startEchoStream(stream EchoRequest) returns (stream EchoResponse) {}
}